lib/pure/times.nim

```diff
@@ -1210,15 +1373,32 @@ proc parse*(value, layout: string, zone: Timezone = local()): DateTime =
         token.add(layout[i])
         inc(i)
       else:
-        parseToken(dt, token, value, j)
+        found_tokens.incl parseToken(result, token, value, j)
         token = ""
 
-  if dt.isDst:
+  if parsedDateComplete * found_tokens != parsedDateComplete:
+      # Assumes current day of month, month and year
+    let dt = date()
+    result.monthday = dt.monthday
+    result.month = dt.month
+    result.year = dt.year
```

Comment:
This is not correct? It will overwrite all of `monthday`, `month` and `year` even if some of them might have been parsed.

You can skip this optimization for now. I've been working on a more robust implementation of format/parse to solve the issues in #7017, as well as other issues I've discovered. The new implementation will contain this optimization.

Reply:
Previous behavior was to call `now()` at the begging of `parse` function unconditionally. It is tearing my heart apart when I see a such wasteful behavior, `now()` is not cheap and it is 95% probably that you will find a date in the string, so results of `now()` call will be overridden and wasted. The idea is to call `now()` only it date was not found in the string